From 28c29dece6c434e7d3d01e805555ba7ddf6fba65 Mon Sep 17 00:00:00 2001
From: "Lu, Han" <han.lu@intel.com>
Date: Wed, 3 Feb 2016 14:20:33 +0800
Subject: [PATCH] alsabat: rename to avoid naming conflict

alsa-utils as well as bareos-bat (as well a some Bacula packages)
all contain a program called /usr/bin/bat, which causes conflicts on
various distributions ("basic audio tester" vs "bareos administration
tool"("bacula administration tool")).
Rename to avoid conflict.

Signed-off-by: Lu, Han <han.lu@intel.com>
Signed-off-by: Takashi Iwai <tiwai@suse.de>

Upstream-Status: Accepted [expected in version 1.1.1]

Signed-off-by: Tanu Kaskinen <tanuk@iki.fi>
---
 bat/Makefile.am          | 10 +++++-----
 bat/{bat.1 => alsabat.1} | 43 ++++++++++++++++++++++---------------------
 bat/bat.c                |  2 +-
 3 files changed, 28 insertions(+), 27 deletions(-)
 rename bat/{bat.1 => alsabat.1} (72%)

diff --git a/bat/Makefile.am b/bat/Makefile.am
index 842ae6b..985f00c 100644
--- a/bat/Makefile.am
+++ b/bat/Makefile.am
@@ -1,9 +1,9 @@
-bin_PROGRAMS = bat
-man_MANS = bat.1
+bin_PROGRAMS = alsabat
+man_MANS = alsabat.1
 
-EXTRA_DIST = bat.1
+EXTRA_DIST = alsabat.1
 
-bat_SOURCES = \
+alsabat_SOURCES = \
 	bat.c \
 	common.c \
 	analyze.c \
@@ -21,4 +21,4 @@ noinst_HEADERS = \
 AM_CPPFLAGS = \
 	      -Wall -I$(top_srcdir)/include
 
-bat_LDADD = @FFTW_LIB@
+alsabat_LDADD = @FFTW_LIB@
diff --git a/bat/bat.1 b/bat/alsabat.1
similarity index 72%
rename from bat/bat.1
rename to bat/alsabat.1
index e00fc27..8d0b9c0 100644
--- a/bat/bat.1
+++ b/bat/alsabat.1
@@ -1,23 +1,23 @@
-.TH BAT 1 "20th October 2015"
+.TH ALSABAT 1 "20th October 2015"
 .SH NAME
-bat \- command\-line sound tester for ALSA sound card driver
+alsabat \- command\-line sound tester for ALSA sound card driver
 
 .SH SYNOPSIS
-\fBbat\fP [\fIflags\fP]
+\fBalsabat\fP [\fIflags\fP]
 
 .SH DESCRIPTION
-\fBBAT(Basic Audio Tester)\fP is a simple command\-line utility intended
-to help automate audio driver and sound server testing with little human
-interaction. BAT can be used to test audio quality, stress test features
-and test audio before and after PM state changes.
+\fBALSABAT(ALSA Basic Audio Tester)\fP is a simple command\-line utility
+intended to help automate audio driver and sound server testing with little
+human interaction. ALSABAT can be used to test audio quality, stress test
+features and test audio before and after PM state changes.
 
-BAT's design is relatively simple. BAT plays an audio stream and captures
-the same stream in either a digital or analog loop back. It then compares
-the captured stream using a FFT to the original to determine if the test
-case passes or fails.
+ALSABAT's design is relatively simple. ALSABAT plays an audio stream and
+captures the same stream in either a digital or analog loop back. It then
+compares the captured stream using a FFT to the original to determine if
+the test case passes or fails.
 
-BAT can either run wholly on the target machine being tested (standalone
-mode) or can run as a client/server mode where by bat client runs on the
+ALSABAT can either run wholly on the target machine being tested (standalone
+mode) or can run as a client/server mode where by alsabat client runs on the
 target and runs as a server on a separate tester machine. The client/server
 mode still requires some manual interaction for synchronization, but this
 is actively being developed for future releases.
@@ -93,8 +93,8 @@ Frequencies with amplitude larger than threshold will be recognized as a
 peak, and the frequency with largest peak value will be recognized as a
 detected frequency.
 .br
-BAT then compares the detected frequency to target frequency, to decide
-if the detecting passes or fails.
+ALSABAT then compares the detected frequency to target frequency, to
+decide if the detecting passes or fails.
 .br
 The default value is 3.0.
 .TP
@@ -117,18 +117,19 @@ Target WAV file to save capture test content.
 .TP
 \fI\-\-local\fP
 Internal loopback mode.
-Playback, capture and analysis internal to BAT only. This is intended for
-developers to test new BAT features as no audio is routed outside of BAT.
+Playback, capture and analysis internal to ALSABAT only. This is intended
+for developers to test new ALSABAT features as no audio is routed outside
+of ALSABAT.
 
 .SH EXAMPLES
 
 .TP
-\fBbat \-P plughw:0,0 \-C plughw:0,0 \-c 2 \-f S32_LE \-F 250\fR
+\fBalsabat \-P plughw:0,0 \-C plughw:0,0 \-c 2 \-f S32_LE \-F 250\fR
 Generate and play a sine wave of 250 Hertz with 2 channel and S32_LE format,
 and then capture and analyze.
 
 .TP
-\fBbat \-P plughw:0,0 \-C plughw:0,0 \-\-file 500Hz.wav\fR
+\fBalsabat \-P plughw:0,0 \-C plughw:0,0 \-\-file 500Hz.wav\fR
 Play the RIFF WAV file "500Hz.wav" which contains 500 Hertz waveform LPCM
 data, and then capture and analyze.
 
@@ -152,7 +153,7 @@ Currently only support RIFF WAV format with PCM data. Please report any bugs to
 the alsa-devel mailing list.
 
 .SH AUTHOR
-\fBbat\fP is by Liam Girdwood <liam.r.girdwood@linux.intel.com>, Bernard Gautier
-<bernard.gautier@intel.com> and Han Lu <han.lu@intel.com>.
+\fBalsabat\fP is by Liam Girdwood <liam.r.girdwood@linux.intel.com>, Bernard
+Gautier <bernard.gautier@intel.com> and Han Lu <han.lu@intel.com>.
 This document is by Liam Girdwood <liam.r.girdwood@linux.intel.com> and Han Lu
 <han.lu@intel.com>.
diff --git a/bat/bat.c b/bat/bat.c
index 086b9fa..ddb60b7 100644
--- a/bat/bat.c
+++ b/bat/bat.c
@@ -272,7 +272,7 @@ static void test_capture(struct bat *bat)
 static void usage(struct bat *bat)
 {
 	fprintf(bat->log,
-_("Usage: bat [-options]...\n"
+_("Usage: alsabat [-options]...\n"
 "\n"
 "  -h, --help             this help\n"
 "  -D                     pcm device for both playback and capture\n"
-- 
2.7.0

